# RNA-Seq Data Analysis in R Workshop

**Date:** November 12th  
**Instructor:** [Simona Merlini](https://merlinis12.github.io/merlinisimona.github.io/)  
**Level:** Basic/Intermediate  
**Requirements:** R, RStudio ([download link](https://rstudio-education.github.io/hopr/starting.html))  

## Overview
RNA-Seq is a widely used method for analyzing gene expression. This workshop focuses on differential expression analysis using R, guiding participants through essential steps from raw count data to identifying differentially expressed genes.

### Objectives
- Normalize RNA-Seq data
- Model fitting and hypothesis testing
- Identify differentially expressed genes
- Hands-on R code examples for independent analysis

## Prerequisites
- Basic knowledge of R and RStudio
- limma package ([link](https://bioconductor.org/packages/release/bioc/html/limma.html))

## Agenda
1. **Introduction to RNA-Seq and Differential Expression**  
   Overview of RNA-seq technology, data structure, and objectives.

2. **Data Preprocessing and Normalization**  
   Steps to preprocess and normalize gene count data.

3. **Differential Expression Analysis**  
   Perform differential expression analysis with DESeq2.

4. **Results Interpretation and Visualization**  
   Practical tips for visualizing and interpreting differential expression results.

## Resources
- [fgsea package for downstream analysis](https://bioconductor.org/packages/release/bioc/html/fgsea.html)
- Example data: *(Provide a link to the demo dataset here)*

For questions or further guidance, feel free to reach out!
